purchase.adoc

```diff
@@ -388,288 +517,533 @@ Fields named "*UUID1" are the standard UUID Type 1 (8-4-4-4-12) representation o
+The `discountValue` field following the `discount` structure contains the total discount amount for the row:
++
+--
+[source,json]
+----
 "discount": {
-  "quantity": 1,
-  "percentage": 10
+    "name": "10% off",
+    "quantity": 1,
+    "percentage": 10
 },
 "discountValue": 1000
-```
+----
+
 *Note:* Currently, we only support one discount per row of items, so the `quantity` field will always be set to 1. This could potentially change in the future.
+--
 
-Cart discounts are expressed using the following JSON structure in the root structure of the purchase. They can also contain either an `amount` or a `percentage` value:
-```json
+* Discounts for the whole purchase are expressed using the following JSON structure.
++
+--
+They can contain either `amount` or `percentage` value.
+
+[source,json]
+----
 "discounts": [
 {
-    "name": "",
     "amount": 2000,
     "quantity": 1
 }
-```
-*Note:* Currently, we only support one discount per cart, so the `discounts` list will only contain one item with `quantity` set to 1. This could potentially change in the future.
-
-See further below for a full response example containing discounts.
-
-#### Example
-Lets observe an example purchase, containing both discounts on rows as well as on the cart itself:
-[options="header"]
-|===
-| Product | Unit price | Amount | Discount
-| Foo | 20 | 2 | 5
-| Bar | 10 | 10 | 10%
-| Cart discount | | | 20%
-|===
-To calculate the total discount, we start by calculating the discount per row. The first row contains 2 item and has a fixed discount amount of 5.
-
-The second row contains 10 items and a percentage discount of 10%:
-
-*Row discounts* = 5 + 10 x 10 x 0.1 = 15.
-
-Then we calculate the cart discount, based on the sum of the previously discounted rows:
+----
 
-*Sum of discounted rows* = 20 x 2 + 10 x 10 - 15 = 125.
+*Note:* Currently, we only support one discount for the whole, so the `discounts` list will only contain one item with `quantity` set to 1. This could potentially change in the future.
 
-*Cart discount* = 125 x 0.2 = 25
-
-So the total discount on this purchase is: 15 + 25 = 40. And the total sum to pay is: 140 - 40 = 100.
-
-A discount can never be greater than the actual amount of the row or cart price.
-
-### How to load all purchases for a user
+See further below for a full response example containing discounts.
+--
 
-You should never retrieve all the purchases in one request, since that can potentially put too much load on the server causing the request to be rejected or time out.
+*Discount calculation example*
 
-Instead, you should request the purchases splitted up into "pages", using the `limit` and `lastPurchaseHash` parameters, with `limit` set to maximum 1000.
+Let's observe an example purchase containing both discounts on rows as well as on the purchase itself.
```

Comment:
This example purchase contains both discounts on rows and on the whole purchase.